data_structures/linked_list/detecting_cycle_in_list.c

```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct Node {
+	int data;
+	struct Node* next;
+};
+
+void push(struct Node** head_ref, int new_data)
+{
+	struct Node* new_node
+		= (struct Node*)malloc(sizeof(struct Node));
+
+	new_node->data = new_data;
+
+	new_node->next = (*head_ref);
+
+	(*head_ref) = new_node;
+}
+
+int detectLoop(struct Node* list)
+{
+	struct Node *slow_p = list, *fast_p = list;
+
+	while (slow_p && fast_p && fast_p->next) {
+		slow_p = slow_p->next;
+		fast_p = fast_p->next->next;
+		if (slow_p == fast_p) {
+			return 1;
+		}
+	}
+	return 0;
+}
+
+int main()
+{
+	struct Node* head = NULL;

```

Comment:
According the https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#test . What do you think about moving your example into the `test` method with asserts?